clang/lib/Driver/Driver.cpp

```diff
@@ -4100,33 +4114,31 @@ class OffloadingActionBuilder final {
 
       // Device compilation generates LLVM BC.
       if (CurPhase == phases::Compile) {
-        for (Action *&A : SYCLDeviceActions) {
-          types::ID OutputType = types::TY_LLVM_BC;
-          if ((SYCLDeviceOnly || Args.hasArg(options::OPT_emit_llvm)) &&
-              Args.hasArg(options::OPT_S))
-            OutputType = types::TY_LLVM_IR;
-          if (SYCLDeviceOnly) {
-            if (Args.hasFlag(options::OPT_fno_sycl_use_bitcode,
-                             options::OPT_fsycl_use_bitcode, false)) {
-              auto *CompileAction =
-                  C.MakeAction<CompileJobAction>(A, types::TY_LLVM_BC);
-              A = C.MakeAction<SPIRVTranslatorJobAction>(CompileAction,
-                                                         types::TY_SPIRV);
-              continue;
+        if (!SYCLTargetInfoList.empty()) {
```

Comment:
Can the conditions on 4116-4117 be merged, now that there's nothing happening beyond this branch?

Reply:
Yes, the conditions can be merged. I have made the change.